packages/mgt-chat/src/statefulClient/BotInfoClient.ts

```diff
@@ -0,0 +1,50 @@
+import { TeamsAppInstallation } from '@microsoft/microsoft-graph-types-beta';
+import { BaseStatefulClient } from './BaseStatefulClient';
+import { loadBotInChat, loadBotIcon } from './graph.chat';
+
+export interface BotInfo {
+  botInfo: Map<string, TeamsAppInstallation>;
+  botIcons: Map<string, string>;
+  loadBotInfo: (chatId: string, appId: string) => Promise<void>;
+}
+
+const requestKey = (chatId: string, appId: string) => `${chatId}::${appId}`;
+
+export class BotInfoClient extends BaseStatefulClient<BotInfo> {
+  private readonly infoRequestMap = new Map<string, Promise<unknown>>();
+
+  public loadBotInfo = async (chatId: string, appId: string) => {
+    const beta = this.betaGraph;
+    if (!beta) return;
+    const key = requestKey(chatId, appId);
+    if (!this.infoRequestMap.has(key)) {
+      const requestPromise = loadBotInChat(beta, chatId, appId);
```

Comment:
You are calling the variable "appId" here, but it appears to actually be the "botId". Since both things exist but are different, I would suggest renaming this.